sympy/physics/mechanics/joint.py

```diff
@@ -737,15 +1068,20 @@ def _generate_speeds(self, speed):
         return speeds
 
     def _orient_frames(self):
-        frame = self._set_orientation()
-        self.child.frame.orient_axis(frame, self.parent_axis, 0)
+        self._joint_axis = self._axis(
+            self.joint_axis, self.parent_interframe, self.child_interframe)
+        axis = self._express_joint_axis(self.parent_interframe)
+        self.child_interframe.orient_axis(self.parent_interframe, axis, 0)
 
     def _set_angular_velocity(self):
-        self.child.frame.set_ang_vel(self.parent.frame, 0)
+        self.child_interframe.set_ang_vel(self.parent_interframe, 0)
 
     def _set_linear_velocity(self):
+        self.child_point.set_pos(self.parent_point, self.coordinates[
+            0] * self.joint_axis.normalize())
         self.parent_point.set_vel(self.parent.frame, 0)
         self.child_point.set_vel(self.child.frame, 0)
-        self.child_point.set_pos(self.parent_point, self.coordinates[0] * self.parent_axis.normalize())
-        self.child_point.set_vel(self.parent.frame, self.speeds[0] * self.parent_axis.normalize())
-        self.child.masscenter.set_vel(self.parent.frame, self.speeds[0] * self.parent_axis.normalize())
+        self.child_point.set_vel(self.parent.frame, self.speeds[
+            0] * self.joint_axis.normalize())
+        self.child.masscenter.set_vel(self.parent.frame, self.speeds[
+            0] * self.joint_axis.normalize())
```

Comment:
I think the below is much more readable formatting.
```suggestion
        self.child.masscenter.set_vel(
            self.parent.frame,
            self.speeds[0] * self.joint_axis.normalize()
        )
```